src/App.tsx

```diff
@@ -21,11 +38,17 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            value={itemsPerPageValue}
           >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            {selectValues.map((value: number) => (
+              <option
+                key={value}
+                value={value}
+                onClick={() => setItemsPerPage(value)}
```

Comment:
```suggestion
```
instead onClickHandler in option set onChange handler on Select which will take event and set event.target.value
it is reason why not possible to change perPage on demo link